collector/admission/boundedqueue.go

```diff
@@ -0,0 +1,136 @@
+package admission
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/google/uuid"
+	orderedmap "github.com/wk8/go-ordered-map/v2"
+)
+
+type boundedQueue struct {
+	maxLimitBytes int64
+	maxLimitWaiters int64
+	currentBytes int64
+	currentWaiters int64
+	mtx sync.Mutex
```

Comment:
nit: would you please name this field `lock`? I think it will help readability.

Reply:
Thanks for the review! Fixed